core/iwasm/common/wasm_runtime_common.c

```diff
@@ -1162,41 +1275,56 @@ wasm_runtime_call_wasm(WASMExecEnv *exec_env,
                        uint32 argv[])
 {
     bool ret = false;
+    uint32 *new_argv = NULL, param_argc, result_argc = 0;
 
     if (!wasm_runtime_exec_env_check(exec_env)) {
         LOG_ERROR("Invalid exec env stack info.");
         return false;
     }
 
 #if WASM_ENABLE_REF_TYPES != 0
-    wasm_runtime_prepare_call_function(exec_env, function);
+    if (!wasm_runtime_prepare_call_function(exec_env, function, argv, argc,
+                                            &new_argv, &param_argc,
+                                            &result_argc)) {
+        LOG_ERROR("Failed to process arugments");
+        return false;
+    }
+#else
+    new_argv = argv;
+    param_argc = argc;
 #endif
 
 #if WASM_ENABLE_INTERP != 0
     if (exec_env->module_inst->module_type == Wasm_Module_Bytecode)
         ret = wasm_call_function(exec_env, (WASMFunctionInstance *)function,
-                                 argc, argv);
+                                 param_argc, new_argv);
 #endif
 #if WASM_ENABLE_AOT != 0
     if (exec_env->module_inst->module_type == Wasm_Module_AoT)
-        ret = aot_call_function(exec_env, (AOTFunctionInstance *)function, argc,
-                                argv);
+        ret = aot_call_function(exec_env, (AOTFunctionInstance *)function,
+                                param_argc, new_argv);
 #endif
+    if (!ret) {
+        return false;
```

Comment:
new_argv might be allocated in wasm_runtime_prepare_call_function but isn't freed here, seems here we should:
```C
if (new_argv != argv)
    wasm_runtime_free(new_argv);
```